react/components/molecules/date-input/date-input.js

```diff
@@ -0,0 +1,162 @@
+    static get propTypes() {
+        return {
+            value: PropTypes.instanceOf(Date),
+            header: PropTypes.string,
+            disabled: PropTypes.bool,
+            activeOpacity: PropTypes.number,
+            onUpdateValue: PropTypes.func,
+            style: ViewPropTypes.style
+        };
+    }
+
+    static get defaultProps() {
+        return {
+            value: new Date(),
+            header: undefined,
+            disabled: false,
+            activeOpacity: 0.75,
+            onUpdateValue: () => {},
+            style: {}
+        };
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            valueData: this.props.value,
+            visible: false
+        };
+    }
+
+    onPress = () => {
+        this.setState({
+            visible: true
+        });
+        if (Platform.OS === "ios") this.container.toggle();
+    };
+
+    onChange = (event, value) => {
+        this.setState(
+            prevState => ({
+                valueData: value || prevState.valueData,
+                visible: false
+            }),
+            () => this.props.onUpdateValue(value)
+        );
+    };
+
+    _style = () => {
+        return [styles.dateInput, this.props.style];
+    };
+
+    _buttonStyle = () => {
+        return [styles.dateInputButton, this.props.disabled ? styles.dateInputButtonDisabled : {}];
+    };
+
+    _renderCalendar = () => {
+        if (Platform.OS === "ios") {
+            return (
+                <ContainerOpenable ref={el => (this.container = el)}>
+                    <DateTimePicker
+                        value={this.state.valueData}
+                        mode={"date"}
+                        is24Hour={false}
+                        display="inline"
+                        onChange={this.onChange}
+                    />
+                </ContainerOpenable>
+            );
+        }
+
+        if (!this.state.visible) return;
+        return (
+            <DateTimePicker
+                value={this.state.valueData}
+                mode={"date"}
+                display={"calendar"}
+                is24Hour={false}
+                onChange={this.onChange}
+            />
+        );
+    };
+
+    render() {
+        return (
+            <View style={this._style()} {...this.id("date-input")}>
+                <Touchable
+                    style={this._buttonStyle()}
+                    activeOpacity={this.props.activeOpacity}
+                    disabled={this.props.disabled}
+                    onPress={this.onPress}
+                >
+                    <View style={styles.columnText}>
+                        {Boolean(this.props.header) && (
+                            <Text style={styles.headerText}>{this.props.header}</Text>
+                        )}
+                        <Text style={styles.dateText}>
+                            {this.state.valueData?.toLocaleDateString()}

```

Comment:
Check if one of https://github.com/ripe-tech/ripe-commons-js/blob/master/js/time.js is more suitable. Relying on the system to localize stuff is always a leap of faith, as we lose control over how it's done.

Reply:
(if robin/storybook don't depend on ripe-commons, feel free to add it)